spec/controllers/school_resgistrations_controller_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'rails_helper'
+
+RSpec.describe Devise::RegistrationsController, type: :controller do
+  FLASH_UPDATE_MSG = "Your account has been updated successfully."
+  FLASH_DELETE_MSG = "Bye! Your account has been successfully cancelled. We hope to see you again soon."
+  before :each do
+    @request.env["devise.mapping"] = Devise.mappings[:school]
+  end
+
+  describe "GET #new" do
+    it "renders the :new view" do
+      get :new
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "GET #edit" do
+    context "signed in school" do
+      it "renders the :edit view" do
+        sign_in create(:school)
+        get :edit
+        expect(response).to render_template(:edit)
+      end
+    end
+    context "no sigend in school" do
+      it "redirects to  the sign_in view" do
+        get :edit
+        expect(response).to redirect_to new_school_session_url
+      end
+    end
+  end
+
+  describe "POST #create" do
+    it "creates a new registration" do
+      expect {
+        post :create, school: attributes_for(:school)
+      }.to change(School,:count).by(1)
+      expect(response).to redirect_to root_url
+    end
+
+    it "does not save the invalid registration" do
+      expect {
+        post :create, school: attributes_for(:invalid_school)
+      }.to_not change(School,:count)
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "PATCH #update" do
+    context "signed in school" do
+      before :each do
+        @school = create(:school)
+        sign_in @school
+      end
+      it 'changes @school registration data' do
+        patch :update, school: { name:"Edited school", current_password:"password"}
```

Comment:
Good practice to have a space after the attribute name `... { name: "Edited school", ... }